go/kbfs/libkbfs/favorites.go

```diff
@@ -273,26 +311,36 @@ func (f *Favorites) closeReq(req *favReq, err error) {
 	req.err = err
 	close(req.done)
 	for _, fav := range req.toAdd {
-		delete(f.inFlightAdds, fav)
+		delete(f.inFlightAdds, fav.Favorite)
 	}
 }
 
 // sendChangesToEditHistory notes any deleted favorites and removes them
 // from this user's kbfsedits.UserHistory.
-func (f *Favorites) sendChangesToEditHistory(oldCache map[Favorite]bool) {
+func (f *Favorites) sendChangesToEditHistory(oldCache map[Favorite]FavoriteData) {
 	for oldFav := range oldCache {
-		if !f.cache[oldFav] {
+		if _, present := f.favCache[oldFav]; !present {
 			f.config.UserHistory().ClearTLF(tlf.CanonicalName(oldFav.Name),
 				oldFav.Type)
 		}
 	}
-	for newFav := range f.cache {
-		if !oldCache[newFav] {
+	for newFav := range f.favCache {
+		if _, present := oldCache[newFav]; !present {
 			f.config.KBFSOps().RefreshEditHistory(newFav)
 		}
 	}
 }
 
+func toFolder(fav Favorite, data FavoriteData) keybase1.Folder {
```

Comment:
Might want to put this as a method on `Favorite`, but if not, maybe rename it to be more specific since this is in a package with a large namespace.  Maybe `favoriteToFolder`?